### PR TITLE
dbExecute

### PR DESCRIPTION
**Pull request by bborgesr**
Moving the relevant code for the `dbExecute()` function to this PR, originally in rstats-db/DBI#108

### CODE REVIEW

File: R/DBConnection.R [hunk 1]
```diff
@@ -106,6 +106,41 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' Execute an SQL statement that does not produce a result set.
+#'
+#' This function should be used when you want to execute a non-
+#' \code{SELECT} query on table (ex: \code{UPDATE}, \code{DELETE},
+#' \code{INSERT INTO}, \code{DROP TABLE}, ...). It will execute
+#' the query and return the number of rows affected by the operation.
+#' (If the return value is 0 or -1, you're using it wrong.)
```
Comment (hadley) [inline R/DBConnection.R]: That seems a bit strong as you might be using it to set sql options, which don't affect any rows.

**Comment by hadley** [comment]
Apart from small doc comment, LGTM.

File: R/DBConnection.R [hunk 2]
```diff
@@ -106,6 +106,40 @@ setGeneric("dbSendQuery",
...
+setGeneric("dbExecute",
+  def = function(conn, statement, ...) standardGeneric("dbExecute")
+)
+
+#' @export
+setMethod("dbExecute", signature("DBIConnection", "character"),
+  function(conn, statement, ...) {
+    rs <- dbSendQuery(conn, statement, ...)
+    on.exit(dbClearResult(rs))
+    dbGetRowsAffected(rs)
```
Comment (krlmlr) [inline R/DBConnection.R]: Should we fix the return type here (numeric, to support >= 2³¹ rows)?

Comment (hadley) [inline R/DBConnection.R]: I think it's reasonable to do that in the docs - I'm not sure if using `valueClass` is good practice. It adds a relatively expensive post-hoc check.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**